src/psyclone/transformations.py

```diff
@@ -3079,13 +3079,11 @@ def _validate(self, node_list):
         super(ACCKernelsTrans, self)._validate(node_list)
 
         # Check that we have at least one loop within the proposed region
-        found = False
         for node in node_list:
             loops = node.walk(Loop)
-            if loops or isinstance(node, Loop):
-                found = True
+            if loops:
                 break
-        if not found:
+        else:
```

Comment:
Is this correct? If so I don't understand it (although pylint is happy)! I would have expected the `else:` to belong with the `if loops` block. Incidentally I think you could simplify further and just do `if node.walk(Loop):`